src/i_winmusic.c

```diff
@@ -1381,43 +1381,27 @@ static boolean I_WIN_InitMusic(void)
 {
     int all_devices;
     int i;
-    MIDIOUTCAPS mcaps;
     MMRESULT mmr;
 
-    // find the midi device that matches the saved one
-    if (winmm_midi_device != NULL)
+    all_devices = midiOutGetNumDevs(); // Does not include MIDI_MAPPER.
+
+    for (i = 0; i < all_devices; i++)
     {
-        all_devices = midiOutGetNumDevs() + 1; // include MIDI_MAPPER
-        for (i = 0; i < all_devices; ++i)
-        {
-            // start from device id -1 (MIDI_MAPPER)
-            mmr = midiOutGetDevCaps(i - 1, &mcaps, sizeof(mcaps));
-            if (mmr == MMSYSERR_NOERROR)
-            {
-                if (strstr(winmm_midi_device, mcaps.szPname))
-                {
-                    MidiDevice = i - 1;
-                    break;
-                }
-            }
+        MIDIOUTCAPS caps;
 
-            if (i == all_devices - 1)
-            {
-                // give up and use MIDI_MAPPER
-                free(winmm_midi_device);
-                winmm_midi_device = NULL;
-            }
+        if (midiOutGetDevCaps(i, &caps, sizeof(caps)) == MMSYSERR_NOERROR &&
+            !strcasecmp(winmm_midi_device, caps.szPname))
```

Comment:
This should be `strncasecmp` since `MIDIOUTCAPS.szPname` is fixed length.

Reply:
I don't think it matters since the comparison ends when the null character is reached.